### ISSUE TITLE
Make it easier to define table structure and afterwards to select which columns to print

### ISSUE THREAD

**Issue by davsclaus** [freva/ascii-table#19]
At Apache Camel we have a CLI that allows to output data in tables, and we want to allow the table to output only specific columns depending on CLI arguments. (for example a --wide mode to list more etc.)

However it is a bit too difficult to make it possible to define the table once with all the columns, alignment, min/max width etc, and then afterwards be able to _select_ which columns to include when printing the table.

So what we have done is to copy the table and have "ugly" if ... elseif ... code in Camel.

Here is a little example whether GAV is enabled or not
https://github.com/apache/camel/blob/main/dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/catalog/CatalogBaseCommand.java#L77

**Comment by freva** [comment]
Thanks for great description. 

I couldn't think of any great way to be able to completely separate column definition and at a later point specify which columns to print since there is no way to reference the columns after the definition, but I added `.visible(boolean)` to `Column` in #20, so you should be to do something like
```
System.out.println(AsciiTable.getTable(AsciiTable.NO_BORDERS, rows, Arrays.asList(
        new Column().header("NAME").dataAlign(HorizontalAlign.LEFT).with(r -> r.name),
        new Column().header("ARTIFACT-ID").dataAlign(HorizontalAlign.LEFT).visible(gav).with(this::shortGav),
        new Column().header("LEVEL").dataAlign(HorizontalAlign.LEFT).with(r -> r.level),
        new Column().header("DESCRIPTION").dataAlign(HorizontalAlign.LEFT).with(this::shortDescription))));
```

Released in 1.7.0

**Comment by davsclaus** [comment]
Thank you very much
